heaps/min_heap.py

```diff
@@ -1,33 +1,44 @@
 class HeapNode:
-  
-    def initialize(self, key, value):
+    def __init__(self, key, value):
         self.key = key
         self.value = value
 
+    def __str__(self):
+        return str(self.value)
 
-class MinHeap:
 
+class MinHeap:
     def __init__(self):
         self.store = []
 
 
     def add(self, key, value = None):
         """ This method adds a HeapNode instance to the heap
             If value == None the new node's value should be set to key
-            Time Complexity: ?
-            Space Complexity: ?
+            Time Complexity: O(log n)
+            Space Complexity: O(1)
         """
```

Comment:
👍 However since heap_up is recursive you have space complexity of O(log n) due to the call stack.